### PR TITLE
Add remove watchlist

### PR DESCRIPTION
**Pull request by danielmolnar**
Implented Watchlist Page with "add to" & "remove from" functionality.

### CODE REVIEW

File: client/src/lib/localStorage.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export const loadFromLocal = (key) => {
+  try {
+    const local = JSON.parse(localStorage.getItem(key));
```
Comment (anitaschiller) [inline client/src/lib/localStorage.js:3]: You could consider summarizing both lines in one: `return JSON.parse(localStorage.getItem(key));`

File: client/src/lib/loadFromLocal.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export default function loadFromLocal(key) {
```
Comment (anitaschiller) [inline client/src/lib/loadFromLocal.js:1]: Seems like you created two loadFromLocal functions - one in lib/loadFromLocal, one in lib/localStorage. If there wasn't a deeper reason behind it, you could delete this js-document here.

File: client/src/components/Overlay.md [hunk 3]
```diff
@@ -0,0 +1,2 @@
+```js
```
Comment (anitaschiller) [inline client/src/components/Overlay.md:1]: I think you forgot to add the component to the ```js part :)

File: client/src/App.js [hunk 4]
```diff
@@ -1,15 +1,67 @@
...
+  const [watchlist, setWatchList] = useLocalStorage('Watchlist', []);
+
+  const isOnWatchList = (movieToAdd) =>
+    watchlist.some((movie) => movie.id === movieToAdd.id);
+
+  function addToWatchList(movieToAdd) {
+    if (!isOnWatchList(movieToAdd)) {
+      setWatchList([...watchlist, movieToAdd]);
+    } else
+      setWatchList(watchlist.filter((movie) => movie.id !== movieToAdd.id));
```
Comment (anitaschiller) [inline client/src/App.js:22]: I'm not sure if I read your code correctly but do you really need the "else"? If the movieToAdd is already on your wishlist simply doing "nothing" could maybe also be an option ;)

File: client/src/App.js [hunk 5]
```diff
@@ -18,6 +70,34 @@ export default App;
 
 const Buffer = styled.div`
   height: 1px;
+  margin-bottom: 100px;
```
Comment (anitaschiller) [inline client/src/App.js:73]: rem might be more practical than px

File: client/src/App.js [hunk 6]
```diff
@@ -18,6 +70,34 @@ export default App;
...
+`;
+
+const WatchlistWrapper = styled.div`
+  display: grid;
+  gap: 1rem;
+  grid-template-columns: repeat(4, 1fr);
+
+  @media (max-width: 800px) {
```
Comment (anitaschiller) [inline client/src/App.js:81]: Cool that you included media queries right away!!

File: client/src/Pages/Friends.js [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import styled from 'styled-components';
+
+export default function Friends() {
+  return <Wrapper></Wrapper>;
```
Comment (anitaschiller) [inline client/src/Pages/Friends.js:5]: Is this page still under construction? I was wondering why it only consists of a Wrapper..

File: client/src/components/AddButton.js [hunk 8]
```diff
@@ -1,17 +1,29 @@
...
+import { Context } from '../hooks/Store';
 
-export default function AddButton() {
-  return <WatchListButton>+ Watchlist</WatchListButton>;
+export default function AddButton({ addToWatchList }) {
+  const [checkWatchList, setCheckWatchList] = useContext(Context);
+
+  return (
+    <WatchListButton onClick={addToWatchList}>
+      {checkWatchList ? 'Remove from ' : 'Add to '}Watchlist
```
Comment (anitaschiller) [inline client/src/components/AddButton.js:11]: Top! :-)

File: client/src/App.js (hunk 4 above)
Comment (danielmolnar) [reply client/src/App.js:22]: The button handles "add" and "remove" at the same time, depending on whether the movie is on the watchlist or not. The "else" line handles the removal. That is why I also have a ternary on the button, which switches from "add to" to "remove from" Watchlist .So 1 button for two functionalities.  ;)

File: client/src/components/Overlay.md (hunk 3 above)
Comment (danielmolnar) [reply client/src/components/Overlay.md:1]: This is not part of the review.

File: client/src/Pages/Friends.js (hunk 7 above)
Comment (danielmolnar) [reply client/src/Pages/Friends.js:5]: Yes it is

File: client/src/App.js (hunk 5 above)
Comment (danielmolnar) [reply client/src/App.js:73]: It is a placeholder, therefore it's called button. As of now, it needs a height for the sake of it being there. 1px is fine. Also what would 1px be in rem?

File: client/src/lib/loadFromLocal.js (hunk 2 above)
Comment (danielmolnar) [reply client/src/lib/loadFromLocal.js:1]: Before I implemented the local storage custom hook, I had two seperate files for "load" and "save". I intended to delete those two, which were no longer needed. Apparently it didnt happen for the "loadFromLocal" file.

File: client/src/lib/localStorage.js (hunk 1 above)
Comment (danielmolnar) [reply client/src/lib/localStorage.js:3]: see comment below :)

**Review by anitaschiller — COMMENTED** [review]
Hi Daniel, 
you did a lot of fancy stuff, well done! Just some minor comments from my side :).

File: client/src/App.js (hunk 4 above)
Comment (anitaschiller) [reply client/src/App.js:22]: Ahh okay, that makes sense - need to clone the repository next time ;).

File: client/src/components/AddButton.js (hunk 8 above)
Comment (danielmolnar) [reply client/src/components/AddButton.js:11]: Thanks :)

**Pull request merged**